### ISSUE TITLE
Missing callable arguments are not detected

### ISSUE THREAD

**Issue by simPod** [vimeo/psalm#3306]
I'm expecting `callable(int, string): string`, passing `function(int $i):string {...}` but psalm does not complain about missing second argument. Shouldn't it? Thanks!

```php
<?php

/**
 * @return callable(
 *    int,
 *    string
 * ): string
 */
function takesAnInt() : callable {
    return function(int $i):string {
        return (string) $i;
    };
}
```
https://psalm.dev/r/82ce984aa3

**Comment by muglug** [comment]
No – `callable(int, string):string` is essentially a subtype of `callable(int):string`
